langchain-core/src/prompts/chat.ts

```diff
@@ -331,6 +334,233 @@ export class ChatMessagePromptTemplate<
   }
 }
 
+interface _TextTemplateParam {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  text?: string | Record<string, any>;
+}
+
+interface _ImageTemplateParam {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  image_url?: string | Record<string, any>;
+}
+
+type MessageClass =
+  | typeof HumanMessage
+  | typeof AIMessage
+  | typeof SystemMessage;
+
+type ChatMessageClass = typeof ChatMessage;
+
+class _StringImageMessagePromptTemplate<
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  RunInput extends InputValues = any,
+  RunOutput extends BaseMessage[] = BaseMessage[]
+> extends BaseMessagePromptTemplate<RunInput, RunOutput> {
+  lc_namespace = ["langchain_core", "prompts", "chat"];
+
+  lc_serializable = true;
+
+  inputVariables: Array<Extract<keyof RunInput, string>> = [];
+
+  additionalOptions: Record<string, unknown> = {};
+
+  prompt:
+    | BaseStringPromptTemplate<
+        InputValues<Extract<keyof RunInput, string>>,
+        string
+      >
+    | Array<
+        | BaseStringPromptTemplate<
+            InputValues<Extract<keyof RunInput, string>>,
+            string
+          >
+        | ImagePromptTemplate<
+            InputValues<Extract<keyof RunInput, string>>,
+            string
+          >
+        | MessageStringPromptTemplateFields<
+            InputValues<Extract<keyof RunInput, string>>
+          >
+      >;
+
+  protected messageClass?: MessageClass;
+
+  // ChatMessage contains role field, others don't.
+  // Because of this, we have a separate class property for ChatMessage.
+  protected chatMessageClass?: ChatMessageClass;
+
+  constructor(
+    /** @TODO When we come up with a better way to type prompt templates, fix this */
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    fields: any,
+    additionalOptions?: Record<string, unknown>
+  ) {
+    if (!("prompt" in fields)) {
+      // eslint-disable-next-line no-param-reassign
+      fields = { prompt: fields };
+    }
+    super(fields);
+    this.prompt = fields.prompt;
+    if (Array.isArray(this.prompt)) {
+      let inputVariables: Extract<keyof RunInput, string>[] = [];
+      this.prompt.forEach((prompt) => {
+        if ("inputVariables" in prompt) {
+          inputVariables = inputVariables.concat(prompt.inputVariables);
+        }
+      });
+      this.inputVariables = inputVariables;
+    } else {
+      this.inputVariables = this.prompt.inputVariables;
+    }
+    this.additionalOptions = additionalOptions ?? this.additionalOptions;
+  }
+
+  createMessage(content: MessageContent) {
+    if (this.messageClass) {
+      return new this.messageClass({ content });
+    } else if (this.chatMessageClass) {
+      // Assuming ChatMessage constructor also takes a content argument
+      return new this.chatMessageClass({
+        content,
+        role: this.getRoleFromMessageClass(this.chatMessageClass.name),
```

Comment:
This will fail in minified environments :( you should be able to use `lc_name` instead?